docs/apis/programming_guide.md

```diff
@@ -187,7 +187,17 @@ that creates the type information for Flink operations.
 </div>
 </div>
 
+#### Scala Dependency Versions
 
+Because Scala 2.10 binary is not compatible with Scala 2.11 binary, we provide multiple artifacts
+to support both Scala versions. If you want to run your program on Flink with Scala 2.11, you need
+to add a suffix `_2.11` to all Flink artifact ids in your dependencies. You should be careful with
+this difference of artifact id. All modules with Scala 2.11 have a suffix `_2.11` in artifact id.
+For example, `flink-java` should be changed to `flink-java_2.11` and `flink-clients` should be
+changed to `flink-clients_2.11`.
```

Comment:
@aalexandrov I'm not sure it is good to use the keyword "Scala-dependent". We cross-build all modules including java-based but linked with Scala modules. I think that omitting "Scala-dependent" is much better.